src/flowcean/transforms/euclidean_distance.py

```diff
@@ -0,0 +1,121 @@
+        data = environment.get_data()
+        transform = EuclideanDistance(
+            feature_a_name="feature_a",
+            feature_b_name="feature_b",
+            output_feature_name="euclidean_distance",
+        )
+        transformed_data = transform.transform(data)
+        ...
+    ```
+
+    The resulting Dataframe after the transform is:
+
+    | feature_a            | feature_b            | euclidean_distance        |
+    | ---                  | ---                  | ---                       |
+    | [{1.0, {1.0, 1.0}},  | [{1.0, {1.0, 2.0}},  | [{1.0, {1.414213562371}}, |
+    |  {2.0, {2.0, 2.0}},  |  {2.0, {1.0, 2.0}},  |  {2.0, {1.414213562371}}, |
+    |  {3.0, {3.0, 3.0}},  |  {3.0, {2.0, 3.0}},  |  {3.0, {1.414213562371}}, |
+    |  {4.0, {4.0, 4.0}}]  |  {4.0, {3.0, 4.0}}]  |  {4.0, {1.414213562371}}] |
+
+    """
+
+    def __init__(
+        self,
+        feature_a_name: str,
+        feature_b_name: str,
+        output_feature_name: str,
+    ) -> None:
+        """Initializes the `EuclideanDistance` transform.
+
+        Args:
+            feature_a_name (str): The name of the first feature.
+            feature_b_name (str): The name of the second feature.
+            output_feature_name (str): The name of the output feature.
+
+        """
+        self.feature_a_name = feature_a_name
+        self.feature_b_name = feature_b_name
+        self.output_feature_name = output_feature_name
+
+    @override
+    def transform(self, data: pl.DataFrame) -> pl.DataFrame:
+        logger.info(
+            "Computing the Euclidean distance between %s and %s.",
+            self.feature_a_name,
+            self.feature_b_name,
+        )
+
+        time = (
+            data.select(self.feature_a_name)
+            .explode(self.feature_a_name)
+            .unnest(self.feature_a_name)
+            .select("time")
+        )
+
+        feature_a = (
+            data.select(self.feature_a_name)
+            .explode(self.feature_a_name)
+            .unnest(self.feature_a_name)
+            .select("value")
+            .unnest("value")
+        )
+        feature_b = (
+            data.select(self.feature_b_name)
+            .explode(self.feature_b_name)
+            .unnest(self.feature_b_name)
+            .select("value")
+            .unnest("value")
+        )
+
+        euclidean_distance = (
+            (feature_a - feature_b)
+            .select(cs.all() ** 2)
+            .with_columns(value=pl.sum_horizontal(cs.all()).sqrt())
+            .select("value")
+        )
+
+        return data.hstack(
+            data.explode(cs.all())
+            .hstack(euclidean_distance)
+            .hstack(time)
+            .select(
+                [
+                    pl.struct(
+                        pl.col("time"),
+                        pl.struct(pl.col("value")),
+                    ).alias(self.output_feature_name)
+                ]
+            )
+            .select(cs.all().implode())
+        )
```

Comment:
Just out of curiosity: Have you benchmarked the code? I think it's much cleaner than the current implementation of `resample`. If it is indeed faster, we might want to apply similar logic to the resample transformation.